src/modules/Modal/Modal.js

```diff
@@ -45,9 +47,18 @@ class Modal extends Component {
     /** Additional classes. */
     className: PropTypes.string,
 
+    /** Icon */
+    closeIcon: PropTypes.oneOfType([
+      PropTypes.node,
```

Comment:
How about allowing boolean as well?  I would think `closeIcon` alone would be the primary use case for this prop.